x-pack/platform/plugins/shared/lens/public/datasources/form_based/esql_layer/text_based_languages.tsx

```diff
@@ -225,7 +226,11 @@ export function getTextBasedDatasource({
     if (fieldName) return [];
     if (context && 'dataViewSpec' in context && context.dataViewSpec.title && context.query) {
       const newLayerId = generateId();
-      const textBasedQueryColumns = context.textBasedColumns?.slice(0, MAX_NUM_OF_COLUMNS) ?? [];
+      const { columns: textBasedQueryColumns } = transformEsqlMultiTermBreakdown({
+        columns: context.textBasedColumns?.slice(0, MAX_NUM_OF_COLUMNS) ?? [],
+        rows: [],
+      });
```

Comment:
If I'm not wrong, at this point you already have the transformed data table, so no need to transform it again

Reply:
It doesn't come through that way... if you don't transform here it retains the original format. The code that's generating the columns here extracts it directly from the ESQL and not the results.